app/src/main/java/com/babylon/wallet/android/presentation/dapp/authorized/login/DAppAuthorizedLoginViewModel.kt

```diff
@@ -193,29 +193,40 @@ class DAppAuthorizedLoginViewModel @Inject constructor(
                 }
                 return Result.success(Unit)
             } else {
-                return Result.failure(RadixWalletException.DappRequestException.InvalidPersonaOrAccounts)
+                return Result.failure(InvalidPersonaOrAccounts)
             }
         }
         return Result.success(Unit)
     }
 
+    @Suppress("ReturnCount")
     private suspend fun validateRequestedAddressesForProofOfOwnership(
         requestedPersonaAddress: IdentityAddress?,
         requestedAccountAddresses: List<AccountAddress>?
     ): Boolean {
-        // if requestedPersonaAddress is present, then check if it exists in the allPersonasOnCurrentNetwork
+        // if both requestedPersonaAddress and requestedAccountAddresses are null then return false
+        if (requestedPersonaAddress == null && requestedAccountAddresses == null) return false
+
+        // validate requestedPersonaAddress if present
         val isIdentityValid = requestedPersonaAddress?.let { address ->
             getProfileUseCase().activePersonasOnCurrentNetwork.any { it.address == address }
-        } ?: true // if requestedPersonaAddress is null, assume it's valid (no need to check)
+        } ?: true // if requestedPersonaAddress is null then it's considered valid
 
-        // if requestedAccountAddresses are present, then check if all of them exist in the existingAccounts
-        val areAccountsValid = requestedAccountAddresses?.let { addresses ->
-            addresses.all { address ->
-                getProfileUseCase().activeAccountsOnCurrentNetwork.any { it.address == address }
-            }
-        } ?: true // if requestedAccountAddresses is null, assume it's valid (no need to check)
+        // if requestedPersonaAddress is not present then return false
+        if (!isIdentityValid) return false
+
+        // if requestedPersonaAddress is valid and requestedAccountAddresses is null then return true
+        if (requestedAccountAddresses == null) return true
+
+        // if requestedPersonaAddress is valid but requestedAccountAddresses is empty then return false
+        if (requestedAccountAddresses.isEmpty()) return false
+
+        // validate requestedAccountAddresses if present and non-empty
+        val areAccountsValid = requestedAccountAddresses.all { address ->
+            getProfileUseCase().activeAccountsOnCurrentNetwork.any { it.address == address }
```

Comment:
I can see the `getProfileUseCase` is called in a loop, but it is has also already been called above at line 212, this means we can easily reuse the returned value from there and avoid redundant IO operations.

Reply:
very good point 👌🏽

Reply:
nit: Similar is`activeAccountsOnCurrentNetwork`, but luckily it's not an IO operation and the impact is negligible.